tests/cloner_test.go

```diff
@@ -531,6 +531,65 @@ var _ = Describe("all clone tests", func() {
 			})
 			ClonerBehavior(cloneStorageClassName, "csivolumeclone")
 		})
+
+		Context("CloneStrategy on storageclass annotation", func() {
+			cloneType := cdiv1.CDICloneStrategy(cdiv1.CloneStrategyCsiClone)
+
+			BeforeEach(func() {
+				if !f.IsCSIVolumeCloneStorageClassAvailable() {
+					Skip("CSI Volume Clone is not applicable")
+				}
+				cloneStorageClassName = f.CsiCloneSCName
+				By(fmt.Sprintf("Get original storage profile: %s", cloneStorageClassName))
+				storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), storageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(storageProfile.Spec.CloneStrategy).To(BeNil())
+				Expect(storageProfile.Status.CloneStrategy).To(BeNil())
+
+				//TODO:  get storageClass `cloneStorageClassName` and annotate with `cloneType`
+				storageProfile.Spec.CloneStrategy = &cloneType
+
+			})
+			AfterEach(func() {
+				By("[AfterEach] Restore the storage class - remove annotation ")
+				// TODO: Remove annotation from storageClass `cloneStorageClassName`
+				storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), storageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				storageProfile.Spec.CloneStrategy = nil
+				storageProfile.Status.CloneStrategy = nil
+				_, err = f.CdiClient.CdiV1beta1().StorageProfiles().Update(context.TODO(), storageProfile, metav1.UpdateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				// TODO: verify that the storageProfile `cloneStorageClassName` has empty storageProfile.Status.CloneStrategy
+				storageProfile, err = f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), storageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(storageProfile.Spec.CloneStrategy).To(BeNil())
+				Expect(storageProfile.Status.CloneStrategy).To(BeNil())
+
+			})
+			It("Should clone  with correct strategy from storageclass annotation ", func() {
+				//TODO: get storageProfile and verify taht the cloneStrategy is set to `cloneType`
+				storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), storageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(storageProfile.Spec.CloneStrategy).ToNot(BeNil())
+				Expect(storageProfile.Status.CloneStrategy).ToNot(BeNil())
+				Expect(*storageProfile.Spec.CloneStrategy).To(Equal(cloneType))
+				Expect(*storageProfile.Status.CloneStrategy).To(Equal(cloneType))
```

Comment:
I think that only `Status` should contain cloneType, `Spec` should be nil

Reply:
We just ensured in the before each that the storage profile is what we want, so no need to check again.

Reply:
I mean you don't need to look up the storage profile here and do the above expects.